### DISCUSSION TITLE
What is the positive and negative edges?

### DISCUSSION THREAD

**Discussion by hkim716** [pyg-team/pytorch_geometric#2102] [Q&A] [answered]
Hi, thanks for the awesome library for GNN. 

When I saw Planetoid dataset, 

`Data(test_neg_edge_index=[2, 527], test_pos_edge_index=[2, 527], train_neg_adj_mask=[2708, 2708], train_pos_edge_index=[2, 8976], val_neg_edge_index=[2, 263], val_pos_edge_index=[2, 263], x=[2708, 1433])`

What do the `pos_edge` and `ned_edge` mean? And what is the positive sampling and negative sampling?
Is the `Neg_adj_mask` equal to the adjacency matrix with shape of [number of nodes, number of nodes]?

**Comment by rusty1s** [comment]
I guess you are looking at the link prediction example. Here, `pos_edge` corresponds to a subset of the original edges in `edge_index`, which are just for training, validation, and testing respectively. `neg_edge_index` corresponds ground-truth negative edges (edges that are not part of the original `edge_index`).

**Reply by hkim716** [reply]
Ok, thanks for your reply. In my understanding, `pos_edge` is used for training, validation, and testing. Then why do we need `neg_edge` even though we have `pos_edge` information, which will be enough for training, validation and testing?

**Comment by rusty1s** [comment, marked as answer]
We need negative edges for training our model, since it is casted as a binary classification problem. You need negative edges to let the model learn how to distinguish positive from negative ones.
